File: Gwonwoo-Nam/season1/week_04/BOJ_3273_두수의합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class BOJ_3273_두수의합 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());



        int[] array = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt)
            .toArray();
        int m = Integer.parseInt(br.readLine());

        Arrays.sort(array);

        int i=0;
        int j=0;
        int count=0;

        while (j < n-1 && array[j] < m) {
            j++;
        }
        while (i<j) {
            if (array[i] + array[j] == m) {
                count++;
                i++;
                j--;
            }
            else if (array[i] + array[j] > m) {
                j--;
            }
            else if (array[i] + array[j] < m) {
                i++;
            }
        }

        System.out.println(count);
    }
}

File: JeonHyoChang/Season1/week_04/BOJ_3273_두수의합.py
import sys

n = int(sys.stdin.readline())
arr = list(map(int, sys.stdin.readline().split()))
arr.sort()
x = int(sys.stdin.readline())

left = 0
right = n - 1
count = 0

while left < right:
    sumOfNum = arr[left] + arr[right]
    if sumOfNum > x:
        right -= 1
    elif sumOfNum < x:
        left += 1
    else:
        left += 1
        right -= 1
        count += 1

print(count)
